serialization.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+func BuildIndex(indexDir string) (*index, error) {
+	if err := os.MkdirAll(indexDir, 0755); err != nil {
```

Comment:
I think you already know this but be careful of the scope of the "if ;" statement.